microsoft/MixedRealityToolkit-Unity
# Fix touch screen controller - part 1

The touch screen controller was previously generating pointing rays in unexpected places (#5420), orphaning some of the rays and not actually generating pointer click events.

To address these, the following changes were made:

* Remove ContactEpsilon as it was overlong (30 sec) since touch screen drivers implement appropriate filtering.
    * The long length of ConactEpsilon was preventing click events from firing (the check was if < 30 sec, else if < .5 sec)
* When setting pose position and rotaion, use the pointer's values and not the base cursor's as thus was causing the pointing rays to be incorrectly positioned.
* Update pointer management as pointers to address rogue rays.

Incorrect behavior:
![image](https://user-images.githubusercontent.com/13281406/63879678-0e561f00-c981-11e9-935c-23b89e7cd1e3.png)

Corrected behavior:
![5420fix](https://user-images.githubusercontent.com/13281406/63882248-7bb87e80-c986-11e9-84a5-c8f8df7236b2.gif)

There are more changes needed to the touch controller and pointer (ex: manipulation does not currently work). These changes will follow in a future PR.

Fixes: #5420 

Validation:
All tests performed on Surface Laptop's touch screen in editor.

[Pre-fix]
* New scene with MRTK object
    * Enter play mode
    * Touch game window with single finger and multiple fingers
    * Confirmed #5420's reported behavior: debug rays (via FocusProvider) were correct, on screen rays (via line renderer) were not
* Load spatial awareness demo
    * Add spatial object mesh observer
    * Enter play mode
    * Attempt to click on the sphere (toggle mesh display) - This did not work
    * Confirm that input simulator properly generates click events

[Post-fix]
* Repeat new scene test
    * Confirmed that rays now point inward from touch point (expected behavior)
    * Confirmed rays (and touch points) are properly cleaned up
* Repeat spatial awareness demo test
    * Confirmed touch click on the sphere functions as expected (properly toggles the mesh display)
